### DISCUSSION TITLE
skip silence speeds up playback at more than 1.0x speeds

### DISCUSSION THREAD

**Discussion by test2a** [PaulWoitaschek/Voice#1876] [Ideas]
attached is the video in a zip file showing 
1. playback is at 1x speed and selecting skip silence does NOT affect playback.
2. playback at any speed more than 1x, here i used 2x as an example. 2x is pretty fine by itself but if i select skip silence, it accelerates audio to some crazy speeds. This should not happen. skip silence should work uniformly at all speeds but here it is not doing that, ie, it should detect long silences (say 1 sec) and accelerate during that period only. 

[1554717037-videorecord-2019-4-8-15-20-37.zip](https://github.com/PaulWoitaschek/Voice/files/3053735/1554717037-videorecord-2019-4-8-15-20-37.zip)

**Comment by rinormaloku** [comment]
Will check this

**Comment by rinormaloku** [comment]
@test2a this looks good to me, the playback speed has not been changed only the silences are being skipped.

I agree that it kinda sounds weird, but I would expect that being on 2x speed and having all silences skipped.

**Comment by PaulWoitaschek** [comment]
I don't understand the issue. Skip silence skips the silence. And faster playback speed plays faster. 
If both are on, the thing is really quick but what else would you expect?

**Comment by ghost** [comment]
> it should detect long silences (say 1 sec) and accelerate during that period only.

It detects 150ms by default https://github.com/google/ExoPlayer/blob/ecb7b8758cd6a700b72275a56d22b3dc56959764/library/core/src/main/java/com/google/android/exoplayer2/audio/SilenceSkippingAudioProcessor.java#L36

**Comment by test2a** [comment]
What I understand from skip silence is to skip out chapter endings and track changes on recordings of old audiobooks which have a lot of "tape static". Currently it skips general word spacing between dialogues and that probably should not happen.

It should only detect longer silences
